### PR TITLE
Fixes following threads design implementation review

### PR DESCRIPTION
**Pull request by gsouquet**
Fixes https://github.com/vector-im/element-web/issues/19646
Fixes https://github.com/vector-im/element-web/issues/19645
Fixes https://github.com/vector-im/element-web/issues/19629

I have not added the display name in the thread summary as we are still waiting to decide on how to tackle disambiguation in that context

Screen Shot 2021-11-09 at 08 09 30
Screen Shot 2021-11-09 at 08 09 49

---
This change is marked as an *internal change* (Task), so will not be included in the changelog.

Preview: https://618cf70ac6796447146e273f--matrix-react-sdk.netlify.app
⚠️ Do you trust the author of this PR? Maybe this build will steal your keys or give you malware. Exercise caution. Use test accounts.

### CODE REVIEW

File: res/css/views/right_panel/_ThreadPanel.scss [hunk 1]
```diff
@@ -131,24 +168,20 @@ limitations under the License.
...
-        border-radius: 8px;
-    }
-
-    .mx_RoomView_messageListWrapper {
+        background: #fff;
         background-color: $background;
-        padding: 8px;
-        border-radius: inherit;
+        border-radius: 8px;
+        width: calc(100% - 16px);
```
Comment (jryans) [inline res/css/views/right_panel/_ThreadPanel.scss:174]: I assume `box-sizing` is no help to remove the `calc`...?

**Review by jryans — APPROVED** [review]
Thanks, looks good overall! 😄

File: res/css/views/right_panel/_ThreadPanel.scss (hunk 1 above)
Comment (gsouquet) [reply res/css/views/right_panel/_ThreadPanel.scss:174]: It created all sort of funky regressions with the scrollbar. It's likely that a more elegant solution exists but I did not manage to figure it out yet

Comment (toger5) [reply res/css/views/right_panel/_ThreadPanel.scss:174]: as reference: https://github.com/matrix-org/matrix-react-sdk/pull/7105
(But gsouquet already found this anyways. Still wanted to link it here again.)

File: res/css/views/right_panel/_ThreadPanel.scss [hunk 2]
```diff
@@ -131,24 +168,20 @@ limitations under the License.
     }
 
     .mx_AutoHideScrollbar {
-        border-radius: 8px;
-    }
```
Comment (toger5) [inline res/css/views/right_panel/_ThreadPanel.scss:135]: What do you think about this solution? https://github.com/matrix-org/matrix-react-sdk/pull/7105
I also fiddled around with this, when I wanted to get the hidden scrollbar with rounded corners working with the [`TimelineCard`](https://github.com/matrix-org/matrix-react-sdk/pull/7112) (Chat card in the right panel). I wanted it to be based on the same solution as the threads so I played around with the threads as well.
```css
.mx_AutoHideScrollbar {
          border-radius: 8px;
          padding-right: 0px;
          -ms-overflow-style: none;  // IE and Edge
          scrollbar-width: none;  // Firefox
          
          &::-webkit-scrollbar {
              display: none; // Chrome, Safari, Opera
          }
}
```
Hiding the scrollbar with the browser and not do the margin/padding trick gave good results with rounded corners.

Comment (gsouquet) [reply res/css/views/right_panel/_ThreadPanel.scss:135]: I don't think that hiding the scrollbar is an option. They are an important indicator to orientate yourself in the app, and  I believe they serve some accessibility purpose too

**Comment by janogarcia** [comment]
@gsouquet Thanks, Germain! 

I've added new issues to https://github.com/vector-im/element-web/issues/19669 and marked with 🔴 **NOT FIXED YET** those that were checked off but aren't really fixed.

**Pull request merged**